### PR TITLE
Add plugin options: `preserve-logging` and `inline-constants`

### LINKED ISSUE #5853: There should be a plugin option for each behavior controlled by `conservative-optimisation`
### Describe the feature you'd like

The `conservative-optimisation` flag implies a few different things. When it is off:
- The float-in pass is more aggressive and may end up increasing the script cost.
- Constants are always inlined, which may end up increasing the script size.
- Builtin applications are always evaluated at compile time, which may cause fewer trace messages to be emitted at runtime.

Currently there's no knob to control any of these things, but there should be for each one of them.

### Describe alternatives you've considered

_No response_

### PR DESCRIPTION
**Pull request by zliu41**
Currently, `conservative-optimisation` implies (or negates) `relaxed-float-in`, `inline-constants` and `preserve-logging`, but only `relaxed-float-in` is a plugin option by itself. This PR adds plugin options for the other two. Closes #5853

### CODE REVIEW

File: doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst [hunk 1]
```diff
@@ -73,6 +73,12 @@
      - Dump Untyped Plutus Core
 
 
+   * - ``inline-constants``
+     - Bool
+     - False
```
Comment (ana-pantilie) [inline doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst:78]: What does this `False` mean here?

Comment (effectfully) [reply doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst:78]: The default value perhaps?

**Review by effectfully — APPROVED** [review]
LGTM

File: doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst (hunk 1 above)
Comment (ana-pantilie) [reply doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst:78]: Shouldn't the default be true or am I confused?

Comment (zliu41) [reply doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst:78]: > Shouldn't the default be true or am I confused?

Right.. I don't know how we got `False` here. I'll need to take a look.

Comment (zliu41) [reply doc/read-the-docs-site/reference/writing-scripts/compiler-options-table.rst]: @ana-pantilie Good catch. Fixed.

**Pull request merged**